### PR TITLE
axgbe: also validate configuration register in GPIO expander

### PR DESCRIPTION
**Pull request by swhite2**
It is possible for a machine to boot into a state in which the configuration register,
responsible for controlling wether an I/O signal is considered an input or output,
contains randomized values. It was assumed this was programmed by the BIOS.

If I/O is reversed, it's possible for the driver to think an SFPP module has been inserted
when there is none, leading to unrecoverable I2C errors.

The configuration register should contain a state which is determined and provided by the BIOS,
hence no hard-coded values are programmed here.

### CODE REVIEW

File: sys/dev/axgbe/xgbe-phy-v2.c [hunk 1]
```diff
@@ -1506,17 +1506,24 @@ xgbe_log_gpio_expander(struct xgbe_prv_data *pdata)
...
+	axgbe_printf(0, "Polarity port registers: 0x%x\n", phy_data->sfp_gpio_polarity);
+	axgbe_printf(0, "Configuration port registers: 0x%x\n", phy_data->sfp_gpio_configuration);
 }
 
 static int
 xgbe_phy_validate_gpio_expander(struct xgbe_prv_data *pdata)
 {
 	struct xgbe_phy_data *phy_data = pdata->phy_data;
-	uint8_t polarity[3];
+	uint8_t gpio_data[3] = {0};
```
Comment (fichtner) [inline sys/dev/axgbe/xgbe-phy-v2.c:1519]: my C is a bit rusty it seems... what's going on here?

Comment (fichtner) [reply sys/dev/axgbe/xgbe-phy-v2.c:1519]: maybe more to the point shouldn't we init all 3 values for safety? or use memset?

Comment (swhite2) [reply sys/dev/axgbe/xgbe-phy-v2.c:1519]: This is an initializer list, it sets all elements to 0: https://stackoverflow.com/questions/2589749/how-to-initialize-array-to-0-in-c

Maybe it doesn't really fit the rest of the code, we can of course use memset - same result :)

Comment (fichtner) [reply sys/dev/axgbe/xgbe-phy-v2.c:1519]: ok if it sets all then that is good 👍

**Review by fichtner — APPROVED** [review]
can't complain, thanks for the fix!

**Pull request merged**